README.adoc

```diff
@@ -1,6 +1,209 @@
+
+While the former method is not publicly available to everyone and requires authentication, we are happy to provide the latter method free of charge and publicly accessible. Please understand that the scripts we provide come without any warranty.
+
+== The Stackable cluster definition YAML
+
+In either way you choose, you have to provide a definition of the Stackable cluster you want to create. You do this in a single YAML file. As of now, this Stackable cluster definition format is work in progress and the only documentation is the following example and the following table:
+
+[source,yaml]
+----
+apiVersion: t2.stackable.tech/v1
+kind: Infra
+template: demo-debian-10
+metadata: 
+  name: stackable-demo
+  description: "This is the cluster I want!"
+domain: stackable.demo
+sshKeys:
+  - "ssh-rsa AAAAB3NzaC1..."
+  - "ssh-rsa AAAACmtp4Ko..."
+spec:
+  region: de/fra
+  nodes:
+    main:
+      numberOfNodes: 1
+      numberOfCores: 2
+      memoryMb: 8192
+      diskType: HDD 
+      diskSizeGb: 15
+    worker:
+      numberOfNodes: 4
+      numberOfCores: 4
+      memoryMb: 8192
+      diskType: HDD 
+      diskSizeGb: 15
+services:
+  spark-primary: |
+    apiVersion: spark.stackable.tech/v1
+    kind: SparkCluster
+    metadata:
+      name: spark-primary
+    spec:
+      master:
+        selectors:
+          - nodeName: "main-1.stackable.demo"
+            instances: 1
+            masterPort: 9999
+            masterWebUiPort: 11111
+      worker:
+        selectors:
+          - nodeName: "worker-1.stackable.demo"
+            instances: 1
+            cores: 1
+            memory: "1g"
+      historyServer:
+        selectors:
+          - nodeName: "worker-3.stackable.demo"
+            instances: 1
+      version: "3.0.1"
+      maxPortRetries: 0
+  spark-secondary: |
+    apiVersion: spark.stackable.tech/v1
+    kind: SparkCluster
+    metadata:
+      name: spark-secondary
+    spec:
+      master:
+        selectors:
+          - nodeName: "main-1.stackable.demo"
+            instances: 1
+            masterPort: 9998
+            masterWebUiPort: 11112
+      worker:
+        selectors:
+          - nodeName: "worker-2.stackable.demo"
+            instances: 1
+            cores: 1
+            memory: "1g"
+      historyServer:
+        selectors:
+          - nodeName: "worker-4.stackable.demo"
+            instances: 1
+      version: "3.0.1"
+      maxPortRetries: 0        
+----
+
+[options="header"]
+|=======
+|key |description
+|apiVersion |always `t2.stackable.tech/v1`
+|kind |always `Infra`
+|template |one of `demo-debian-10` or `demo-centos-7` depending on the desired OS
+|metadata.name |name of the cluster
+|metadata.description |description of the cluster
+|domain |domain for DNS inside the cluster or when accessing through VPN
+|sshKeys |list of SSH public keys to allow access to cluster nodes
+|spec.region |one of the regions that IONOS provides
```

Comment:
If I am not mistaken by adding different templates we could use other providers than IONOS? If so, maybe stay "neutral" here.

Reply:
To be honest, I am not sure if merely a change in the template would allow us to use other providers. Maybe there will be a code change required as well (not a huge one though). But I agree that we should be neutral in our definition file. I assume we'll end up with a subsection that contains cloud-vendor-specific fields.